hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -939,6 +980,290 @@ private void populateCompactionDAG(List<String> inputFiles,
+    AtomicReference<SnapshotLogInfo> snapshotLogInfo =
+        new AtomicReference<>();
+    try (Stream<String> logStream = Files.lines(compactionLogFile, UTF_8)) {
+      logStream.forEach(logLine -> {
+        if (!logLine.startsWith(COMPACTION_LOG_SEQ_NUM_LINE_PREFIX)) {
+          return;
+        }
+
+        snapshotLogInfo.set(getSnapshotLogInfo(logLine));
+      });
+    } catch (IOException exception) {
+      throw new RuntimeException("Failed to read compaction log file: " +
+          compactionLogFile, exception);
+    }
+
+    return snapshotLogInfo.get();
+  }
+
+  /**
+   * Converts a snapshot compaction log line to SnapshotLogInfo.
+   */
+  private SnapshotLogInfo getSnapshotLogInfo(String line) {
+    String[] splits = line.split(" ");
+    assert (splits.length == 4);
+
+    return new SnapshotLogInfo(Long.parseLong(splits[1]),
+        splits[2],
+        Long.parseLong(splits[3]));
+  }
+
+  /**
+   * Returns the list of SST files got compacted in the last compaction from
+   * the provided list of compaction log files.
+   * We can't simply use last file from the list because it is possible that
+   * no compaction happened between the last snapshot and previous to that.
+   * Hence, we go over the list in reverse order and return the SST files from
+   * first the compaction happened in the reverse list.
+   * If no compaction happen at all, it returns empty list.
+   */
+  @VisibleForTesting
+  List<String> getLastCompactionSstFiles(
+      List<Path> compactionLogFiles
+  ) {
+
+    if (compactionLogFiles.isEmpty()) {
+      return Collections.emptyList();
+    }
+    compactionLogFiles = new ArrayList<>(compactionLogFiles);
+    Collections.reverse(compactionLogFiles);
+
+    for (Path compactionLogFile: compactionLogFiles) {
+      List<String> sstFiles = getLastCompactionSstFiles(compactionLogFile);
+      if (!sstFiles.isEmpty()) {
+        return  sstFiles;
+      }
+    }
+
+    return Collections.emptyList();
+  }
+
+  private List<String>  getLastCompactionSstFiles(Path compactionLogFile) {
+
+    AtomicReference<String> sstFiles = new AtomicReference<>();
+
+    try (Stream<String> logStream = Files.lines(compactionLogFile, UTF_8)) {
+      logStream.forEach(logLine -> {
+        if (!logLine.startsWith(COMPACTION_LOG_ENTRY_LINE_PREFIX)) {
+          return;
+        }
+        sstFiles.set(logLine);
+      });
+    } catch (IOException exception) {
+      throw new RuntimeException("Failed to read file: " + compactionLogFile,
+          exception);
+    }
+
+    String lastCompactionLogEntry = sstFiles.get();
+
+    if (StringUtils.isEmpty(lastCompactionLogEntry)) {
+      return Collections.emptyList();
+    }
+
+    // Trim the beginning
+    lastCompactionLogEntry = lastCompactionLogEntry
+        .substring(COMPACTION_LOG_ENTRY_LINE_PREFIX.length());
+
+    String[] io = lastCompactionLogEntry
+        .split(COMPACTION_LOG_ENTRY_INPUT_OUTPUT_FILES_DELIMITER);
+
+    assert (io.length == 2);
```

Comment:
ICYDK `assert`s are ignored by JVM unless explicitly enabled with arg `-ea`. If that is not desired, use Guava `Preconditions.checkArgument` instead.

Reply:
I was not aware of it. I wrote test and it was failing and thought it should work. I'll change it to Guava's `Preconditions.check`.